test/cpp/api/nn_utils.cpp

```diff
@@ -101,3 +101,41 @@ TEST_F(NNUtilsTest, ClipGradNorm) {
     ASSERT_TRUE(torch::allclose(p1.grad(), p2.grad()));
   }
 }
+TEST_F(NNUtilsTest, ClipGradValue) {
+  auto linear_layer = Linear(10, 10);
+  float clip_value = 2.5;
+
+  torch::Tensor grad_w = torch::arange(-50., 50).view({10, 10}).div_(5);
+  torch::Tensor grad_b = torch::ones(10).mul_(2);
+  std::vector<std::vector<torch::Tensor>> grad_lists = {
+      {grad_w, grad_b}, {grad_w, torch::Tensor()}};
+  for (auto grad_list : grad_lists) {
+    for (int i = 0; i < grad_list.size(); i++) {
+      linear_layer->parameters()[i].grad() = grad_list[i].defined()
+          ? grad_list[i].clone().view_as(linear_layer->parameters()[i].data())
+          : grad_list[i];
```

Comment:
I feel that it might be easier to read if we write:
```cpp
auto p = linear_layer->parameters()[i];
auto g = grad_list[i];
p.grad() = g.defined() ? g.clone().view_as(p.data()) : g;
```
which matches the Python version more closely as well :D